packages/cli-core/src/format/format.ts

```diff
@@ -9,19 +9,38 @@ import {
   red,
   underline,
 } from 'kleur/colors';
+import { PackageManagerController } from '@aws-amplify/plugin-types';
+import { PackageManagerControllerFactory } from '../package-manager-controller/package_manager_controller_factory.js';
+import { AmplifyFault } from '@aws-amplify/platform-core';
+
+/**
+ * Exported for testing only. Should not be used directly outside of tests!
+ */
+export const packageManagerControllerContainer: {
+  packageManagerController?: PackageManagerController;
+} = {};
 
 /**
  * Formats various inputs into single string.
  */
-export const format = {
-  runner: (binaryRunner: string) => ({
-    amplifyCommand: (command: string) => {
-      if (command === '') {
-        throw new Error('Command cannot be empty');
-      }
-      return cyan(`${binaryRunner} amplify ${command}`);
-    },
-  }),
+export const format: Format = {
+  backendCliCommand: (command: string) => {
+    if (command.length === 0) {
+      throw new AmplifyFault('InvalidFormatFault', {
+        message: 'The command must be non-empty',
+      });
+    }
+    if (!packageManagerControllerContainer.packageManagerController) {
+      packageManagerControllerContainer.packageManagerController =
+        new PackageManagerControllerFactory().getPackageManagerController();
+    }
+    return cyan(
+      packageManagerControllerContainer.packageManagerController.getCommand([
+        'amplify',
+        ...command.split(' '),
+      ])
+    );
+  },
```

Comment:
How about instead of `packageManagerControllerContainer` we create `createFormat/getFormat` that is package private (exported internally) and do.
`const format:Format = createFormat(new PackageManagerControllerFactory().getPackageManagerController())`.
That way `createFormat` should open a way to test it.

Reply:
The problem is having a const definition that initializes the package manager controller on import. We have to have some way of lazily initializing it only when `backendCliCommand` is called

Reply:
ah I see .

Reply:
Let me experiment a bit with decomposing the package manager controller a bit, because all we really need here is the name of the executable. We don't need all the other package manager baggage.